### ISSUE TITLE
Add sync api for S3 filesystem 

### ISSUE THREAD

**Issue by mmartsen** [PrefectHQ/prefect#6457]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the proposed behavior

Add sync version of filesystem implementation

### Describe the current behavior

Currently prefect.filesystems package includes only async versions of the methods. It basically forcing to switch whole flow to asyncio even if there is no other need in that.

**Comment by madkinsz** [comment]
This should be as easy as adding `@sync_compatible` to the primary interface methods.

**Comment by yashlad681** [comment]
Hi, please dont mind my noob question but I would like to work on this issue but I am not getting exactly what 
`This should be as easy as adding @sync_compatible to the public interface methods.`  means?

Do i have to write an api or is it documentation ? 
Would like to know 
Thanks

**Comment by madkinsz** [comment]
We've written a decorator that let's an async function be run in a sync context. You can see some example usage here: https://github.com/PrefectHQ/prefect/search?q=sync_compatible

Basically, that decorator just needs to be included on the file system functions that a user would call from a synchronous context. Then, a couple tests need to be added covering synchronous usage of those functions.

**Comment by yashlad681** [comment]
Okay, 
Based on your comments

Do you mean in this .py file
https://github.com/PrefectHQ/prefect/blob/ea6f062d5005096ff890aab9595cecc1e4bde9ed/src/prefect/utilities/asyncutils.py

 decorator is supposed to be added fort all the similar below functions?

![image](https://user-images.githubusercontent.com/30846820/185441357-056a12d3-d62a-42e9-a0e3-1b28076002c4.png)

**Comment by madkinsz** [comment]
Here's an example adding it to a couple methods in the `RemoteFileSystem`

```python
diff --git a/src/prefect/filesystems.py b/src/prefect/filesystems.py
index 312431dfd..6d5681ce3 100644
--- a/src/prefect/filesystems.py
+++ b/src/prefect/filesystems.py
@@ -12,7 +12,7 @@ import fsspec
 from pydantic import Field, SecretStr, validator
 
 from prefect.blocks.core import Block
-from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from prefect.utilities.asyncutils import run_sync_in_worker_thread, sync_compatible
 from prefect.utilities.filesystem import filter_files
 
 
@@ -293,6 +293,7 @@ class RemoteFileSystem(ReadableFileSystem, WritableFileSystem):
             counter += 1
         return counter
 
+    @sync_compatible
     async def read_path(self, path: str) -> bytes:
         path = self._resolve_path(path)
 
@@ -301,6 +302,7 @@ class RemoteFileSystem(ReadableFileSystem, WritableFileSystem):
 
         return content
 
+    @sync_compatible
     async def write_path(self, path: str, content: bytes) -> str:
         path = self._resolve_path(path)
         dirpath = path[: path.rindex("/")]
```

**Comment by yashlad681** [comment]
Okay this gives more help .. so again noob question I have to add this decorator to all the functions inside of those 8 classes in prefect.filesystems right? 
I see read_path and write_path is present in almost all of them

**Comment by sibhy** [comment]
Hi is this still open? I'd like to work on this if so. I'm a bit new so I'm planning to follow the instructions from the discussion above. Thanks!

**Comment by madkinsz** [comment]
Ah no it looks like this was implemented in #6511 — sorry about that. Welcome to the repository though! We're happy to help out so feel free to check in on another one that interests you :)
